### PR TITLE
Convert heading to yaw for mag field generation

### PR DESCRIPTION
**Pull request by dlavoieb**

### CODE REVIEW

File: libraries/AP_Compass/AP_Compass_GPS.cpp [hunk 1]
```diff
@@ -82,6 +82,50 @@ void AP_Compass_GPS::_setup_earth_field(void)
...
+
+        /*
+        // Calculate yaw from heading, pitch and roll data from the GPS.
+        // Calculate the Euler form of the pitch and roll data.
+        roll.from_euler(roll, 0, 0);
+        pitch.from_euler(0,pitch,0);
+        */
+
+        // Set Earth Magnetic field
+        _setup_earth_field();
```
Comment (dlavoieb) [inline libraries/AP_Compass/AP_Compass_GPS.cpp]: This isn't used here, maybe remove it

File: libraries/AP_Compass/AP_Compass_GPS.cpp [hunk 2]
```diff
@@ -82,6 +82,50 @@ void AP_Compass_GPS::_setup_earth_field(void)
...
+        // As such, tan(true_heading) to find the ratio
+        // The magneticField is in vector3f form
+        float ratio = tan(heading);
+        
+        // As we need:
+        // y_axis = Ratio[Earth_mag_x*cos(yaw)cos(pitch) + Earth_mag_y*sin(yaw)cos(pitch) + Earth_mag_z*(-sin(pitch))]
+        // y_axis = [Earth_mag_x*cos(yaw)sin(pitch)sin(roll)-sin(yaw)cos(roll)] + [Earth_mag_y*sin(yaw)sin(roll)+cos(yaw)cos(roll)] + [Earth_mag_z*cos(pitch)sin(roll)]
+        // N.B.: FOR THE MOMENT, WITH THREE MAGNETIC EARTH VALUE THIS IS NOT POSSIBLE. AS SUCH, ONLY 400 AT THE X EARTH MAGNETIC
+        // IS BEING USED FOR A MORE SIMPLE YAW CALCULATION
+        yaw = atan2(-ratio*cosf(pitch)-(sinf(pitch)*sin(roll)), cos(roll));
```
Comment (dlavoieb) [inline libraries/AP_Compass/AP_Compass_GPS.cpp]: Why are some function `sinf` and others `sin`?

Comment (junyshin) [inline libraries/AP_Compass/AP_Compass_GPS.cpp]: Oups made a mistake. It should all be sinf.

File: libraries/AP_Compass/AP_Compass_GPS.cpp (hunk 1 above)
Comment (junyshin) [inline libraries/AP_Compass/AP_Compass_GPS.cpp]: I will remove it.

**Pull request merged**